### ISSUE TITLE
Return the proper status codes from the WebAPI

### ISSUE THREAD

**Issue by XVincentX** [ExpressGateway/express-gateway#593]
If not for some particular exceptions, the status code from our WebAPI is always 500, regardless of the kind of error.

If an object already exists, for example, it should be returned as `409`. If the object does not pass the schema validation, it should be a `415`.

Go through the whole WebAPI and verify that the status codes are being returned correctly.

Hint: `Error` class's got a `name` property as well that can be used to identify the error type.

**Comment by iwko** [comment]
I can work on it.

**Comment by XVincentX** [comment]
That'd be great. Do you know where to look at or do you need any guidance?

**Comment by iwko** [comment]
@XVincentX `/lib/rest/` is the place where I should make changes?

**Comment by XVincentX** [comment]
Correct. One example would be [here](https://github.com/ExpressGateway/express-gateway/blob/master/lib/rest/routes/users.js#L14)

**Comment by iwko** [comment]
@XVincentX please assign me

**Comment by XVincentX** [comment]
Go for it @iwko

**Comment by iwko** [comment]
@XVincentX  I went through `/lib/rest/` and I am not sure I am able to find wrong exceptions. https://github.com/iwko/express-gateway/commit/a58efdcdbf373c6c7e9bec276d52e8a2e01fcce3 I did example commit and I am not sure if this is what you had in mind?

**Comment by XVincentX** [comment]
@iwko Not exactly that.

https://github.com/ExpressGateway/express-gateway/blob/master/lib/rest/routes/users.js#L8-L17

You can see here that, in case the schema validation fails, a generic error is returned using `next(err)`. On the other hand, the end user should definitely know that the thing didn't go through because of a schema validation error. 

Does that clarify the issue?

**Comment by iwko** [comment]
Okay, now I've got the point

**Comment by XVincentX** [comment]
If you need any clarification or help feel free to write me. There are some parts that might need to be refactored in order to make that happen; I can take care of those eventually.

**Comment by iwko** [comment]
@XVincentX I think I'm gonna pass on this issue. I've tried to figure out these codes but I am not sure which methods should have statuses added

**Comment by XVincentX** [comment]
That makes sense — it might be a bit complicated and tiring to do that. I'll handle it.

Thanks anyway!

**Comment by shashankkumar8558** [comment]
is this resoved ? @XVincentX
